### PR TITLE
PYTHON-3830 PyMongoCrypt Depends on Deprecated Distutils

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/pymongocrypt/binding.py [hunk 1]
```diff
@@ -1443,6 +1435,14 @@ def __getattr__(self, name):
     lib = _Library(exc)
 else:
     # Check the libmongocrypt version when the library is found.
+    def _parse_version(version_str):
+        # libmongocrypt versions start with x.y.z and have an optional
+        # hypen (e.g. 1.8.0 or 1.8.5-alpha1)
+        if '-' in version_str:
+            version_str = version_str[version_str.index('-')]
```
Comment (ShaneHarvey) [inline bindings/python/pymongocrypt/binding.py:1442]: version_str will always be `"-"` after this line.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/pymongocrypt/binding.py (hunk 1 above)
Comment (blink1073) [reply bindings/python/pymongocrypt/binding.py]: Fixed

**Comment by ShaneHarvey** [comment]
Tests are still failing:
```
[2023/07/14 19:42:03.376] FAIL: test_parse_version (test.test_binding.TestBinding)
[2023/07/14 19:42:03.376] ----------------------------------------------------------------------
[2023/07/14 19:42:03.376] Traceback (most recent call last):
[2023/07/14 19:42:03.376]   File "/data/mci/adfa06c2e1e1064c5cef0c446fd81917/libmongocrypt/bindings/python/test/test_binding.py", line 59, in test_parse_version
[2023/07/14 19:42:03.376]     self.assertLess(_parse_version(v),
[2023/07/14 19:42:03.376] AssertionError: [1, 1, 0] not less than [1, 1, 0]
```

**Comment by blink1073** [comment]
I switched to using the parser from `packaging`.

**Comment by blink1073** [comment]
Oh wow, failure on Windows during self install on Python 12:

```
 [2023/07/14 20:16:33.482] Windows fatal exception: access violation
 [2023/07/14 20:16:33.482] Current thread 0x0000065c (most recent call first):
 [2023/07/14 20:16:33.482] Windows fatal exception: access violation
```

**Comment by blink1073** [comment]
Running this locally on Windows I am seeing:

```python
$ python -m unittest test/test_binding.py
Windows fatal exception: access violation

Current thread 0x0000117c (most recent call first):
  <no Python frame>
```

**Comment by blink1073** [comment]
Looking at https://foss.heptapod.net/pypy/cffi/-/issues/566, `cffi` does not yet support Python 3.12, that would be my guess.

File: bindings/python/pymongocrypt/version.py [hunk 2]
```diff
@@ -14,4 +14,4 @@
 
 __version__ = '1.6.2.dev0'
 
-_MIN_LIBMONGOCRYPT_VERSION = '1.8.0'
+_MIN_LIBMONGOCRYPT_VERSION = '1.5.9'
```
Comment (ShaneHarvey) [inline bindings/python/pymongocrypt/version.py:17]: Let's revert this change.

File: bindings/python/.evergreen/test.sh [hunk 3]
```diff
@@ -19,7 +19,8 @@ if [ "Windows_NT" = "$OS" ]; then # Magic variable in cygwin
              "C:/python/Python38/python.exe"
              "C:/python/Python39/python.exe"
              "C:/python/Python310/python.exe"
-             "C:/python/Python311/python.exe")
+             "C:/python/Python311/python.exe"
+             "C:/python/Python312/python.exe")
```
Comment (ShaneHarvey) [inline bindings/python/.evergreen/test.sh:23]: Let's remove this until cffi works on 3.12 otherwise the build will be failing for everyone.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/pymongocrypt/version.py (hunk 2 above)
Comment (blink1073) [reply bindings/python/pymongocrypt/version.py]: Done

File: bindings/python/.evergreen/test.sh (hunk 3 above)
Comment (blink1073) [reply bindings/python/.evergreen/test.sh]: Switched to testing on RHEL

**Review by ShaneHarvey — APPROVED** [review]
LGTM! I opened https://jira.mongodb.org/browse/PYTHON-3849 for the utcnow deprecation warnings.
